src/app/plugins/tictactoe.plugin.ts

```diff
@@ -311,10 +265,15 @@ class TTTGame {
     this._currentPlayer *= -1;
   }
 
-  private _checkAndUpdateWin() {
+  private _updateGameState() {
     if (this._checkWin()) {
-      this._winner = this._currentPlayer;
-      this._gameOver = true;
+      this.winner = this._convertPlayerToUser(this._currentPlayer);
+      this.loser = this._convertPlayerToUser(this._currentPlayer * -1);
```

Comment:
Perhaps we can use an enum and leverage the type system:

```typescript
enum ResultType {
  won,
  tied,
}
...

type GameResult = { type: ResultType.tied } | { type: ResultType.won, user: User }

// Method
public get result(): GameResult;

...

const result = game.result;

// Checking
switch(result.type) {
  case ResultType.won:
    // We don't need to check for `undefined` on `result.user` here, because typescript's
    // type system is smart enough to figure out that `user` can *never* be undefined when
    // when result.type == ResultType.won.
    console.log(result.user.name);
  case ResultType.tie:
    // This is compile time error, same process as above but ts knows that
    // if the result type == .tied user doesn't exist.
    console.log(result.user);
}
```

Reply:
bump